torch/csrc/lazy/core/shape.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <atomic>
+#include <ostream>
+#include <vector>
+
+#include <c10/core/Scalar.h>
+#include <c10/util/Logging.h>
+
+namespace lazy_tensors {
+
```

Comment:
not sure if you saw my message somewhere else, but can you also put the ShapeIndex class (From shape_util.h) here?  I moved it into shape.h in my big PR but that hasn't landed to staging yet.  ShapeIndex will be a dep of lowering_context unless we refactor that, but it's also pretty small and unoffensive looking